### PR TITLE
Logging and pipeline changes to address failures to send data to DBM endpoints

### PR DESCRIPTION
**Pull request by EmmaOnThursday**
### What does this PR do?
This change doubles the defaultBatchMaxContentSize for the dbm-samples pipeline, and adds debug logging when we reject a message due to content size. 

### Motivation

We have been seeing some issues where data fails to send
to our dbm samples endpoint. We see far fewer examples of
the dbm-activity and dbm-metrics pipelines being blocked or
failing to send data, and we suspect that this is partly
due to the drastic differences in size between those
pipelines and the dbm-samples pipeline.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by Kaderinho — APPROVED** [review]

File: pkg/logs/client/http/destination.go [hunk 1]
```diff
@@ -295,8 +297,13 @@ func (d *Destination) unconditionalSend(payload *message.Payload) (err error) {
 	if err != nil {
 		// the read failed because the server closed or terminated the connection
 		// *after* serving the request.
+		log.Debugf("Server closed or terminated the connection after serving the request with err %v", err)
 		return err
 	}
+
+	metrics.DestinationHttpRespByStatus.Add(strconv.Itoa(resp.StatusCode), 1)
+	metrics.TlmDestinationHttpRespByStatus.Inc(strconv.Itoa(resp.StatusCode))
```
Comment (jmeunier28) [inline pkg/logs/client/http/destination.go:305]: will we know which endpoint is failing to send by recording this? 


Also, should we be recording _something_ when we [hit network issues](https://github.com/DataDog/datadog-agent/pull/15058/files#diff-9e74f15ec100d26628739bfd3f6dd33d24fda6af1c843993a21b268ec97b4c1eR287-R292) aka the context gets canceled & we retry?

File: pkg/logs/client/http/destination.go [hunk 2]
```diff
@@ -218,11 +220,11 @@ func (d *Destination) sendAndRetry(payload *message.Payload, output chan *messag
 		if err != nil {
 			metrics.DestinationErrors.Add(1)
 			metrics.TlmDestinationErrors.Inc()
+			log.Warnf("Could not send payload: %v", err)
```
Comment (jmeunier28) [inline pkg/logs/client/http/destination.go:223]: was this log line moved up intentionally? (seems fine, just wondering)

Comment (EmmaOnThursday) [reply pkg/logs/client/http/destination.go:223]: yeah, previously we were only logging if we got a context canceled error specifically; now we'll log any time we see an error.

Comment (jmeunier28) [reply pkg/logs/client/http/destination.go:223]: I wonder how often we even see errors that aren't timeouts 🤷 

hopefully never

**Review by gh123man — APPROVED** [review]

File: pkg/logs/client/http/destination.go (hunk 1 above)
Comment (EmmaOnThursday) [reply pkg/logs/client/http/destination.go:305]: i can add the url as a tag to this metric, but we don't have clean endpoint or pipeline names at this point. 

for some reason this link just drops me into the page of all files changed, are you pointing to specific code location?

Comment (jmeunier28) [reply pkg/logs/client/http/destination.go]: Oh sorry, yeah the context cancellation is being caught above this code & then we re-try so im wondering if we want to record these timeouts happening

Comment (EmmaOnThursday) [reply pkg/logs/client/http/destination.go]: ah we actually do record those already when the error is returned in sendAndRetry: https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/client/http/destination.go#L218-L224

Comment (jmeunier28) [reply pkg/logs/client/http/destination.go]: Can you show me where the metrics are that record the timeout?? Didn't know that existed

**Pull request merged**